src/systems/triggered_publisher/TriggeredPublisher.cc

```diff
@@ -598,31 +612,73 @@ void TriggeredPublisher::Configure(const Entity &,
   else
   {
     ignerr << "No ouptut specified" << std::endl;
-    return;
+  }
+
+  if (sdfClone->HasElement("service"))
+  {
+    for (auto serviceElem = sdfClone->GetElement("service"); serviceElem;
+         serviceElem = serviceElem->GetNextElement("service"))
+    {
+      ServiceOutputInfo serviceInfo;
+      serviceInfo.servName = serviceElem->Get<std::string>("name");
+      if (serviceInfo.servName.empty())
+      {
+        ignerr << "Service name cannot be empty\n";
+        return;
+      }
+      serviceInfo.reqType = serviceElem->Get<std::string>("reqType");
+      if (serviceInfo.reqType.empty())
+      {
+        ignwarn << "Service request type is empty\n";
+      }
+      serviceInfo.repType = serviceElem->Get<std::string>("repType");
+      if (serviceInfo.repType.empty())
+      {
+        ignwarn << "Service reply type is empty\n";
+      }
+      serviceInfo.reqMsg = serviceElem->Get<std::string>("reqMsg");
+      if (serviceInfo.reqMsg.empty())
+      {
+        ignwarn << "Service request string is empty\n";
+      }
```

Comment:
For all of these, what happens exactly if if they are empty?  Since there is no `continue` after each of these checks, that means that the (partially-unconfigured?) `serviceInfo` is being pushed back into the `std::vector`.  What happens when a request comes in then?  Does it actually work?

Reply:
So these were to account for scenarios like below (snippet from the test). One plugin doesn't have  `repType` (line 340) and the other plugin doesn't have `reqType` (line 351). There are cases when we want to call the `Request` method with no `request Type` or `reply Type`. When a request comes in without a `repType`, it'll call the `Request` method with the right signature
```
333     <plugin$                                                                    
334       filename="ignition-gazebo-triggered-publisher-system"$                    
335       name="ignition::gazebo::systems::TriggeredPublisher">$                    
336       <input type="ignition.msgs.Empty" topic="/in_23"/>$                       
337       <service$                                                                 
338         name="/srv-3"$                                                          
339         reqType="ignition.msgs.StringMsg"$                                      
340         repType=""$                                                             
341         reqMsg="data: 'srv-3'">$                                                
342       </service>$                                                               
343     </plugin>$                                                                                                                               
344 $                                                                               
345     <plugin$                                                                    
346       filename="ignition-gazebo-triggered-publisher-system"$                    
347       name="ignition::gazebo::systems::TriggeredPublisher">$                    
348       <input type="ignition.msgs.Empty" topic="/in_24"/>$                       
349       <service$                                                                 
350         name="/srv-4"$                                                          
351         reqType=""$                                                             
352         repType="ignition.msgs.StringMsg">$                                     
353       </service>$                                                               
354     </plugin>$    
```